sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -2434,6 +2668,43 @@ def load_media_settings(self):
         with open(media_settings_file_path, "r") as media_file:
             g_dict = json.load(media_file)
 
+    """
+    Initializes CMIS_REINIT_REQUIRED and NPU_SI_SETTINGS_SYNC_STATUS fields in APPL_DB:PORT_TABLE
+    if not already present for a port.
+    """
+    def initialize_port_init_fields_in_port_table(self, port_mapping_data):
+        logical_port_list = port_mapping_data.logical_port_list
+        for lport in logical_port_list:
+            asic_index = port_mapping_data.get_asic_id_for_logical_port(lport)
+            appl_db = self.xcvr_table_helper.get_non_producer_app_port_tbl(asic_index)
+            if appl_db is None:
+                helper_logger.log_error("appl_db is None for lport {} during init".format(lport))
+                continue
+
+            app_port_table_fvs_dict = appl_db.get_all(appl_db.APPL_DB, 'PORT_TABLE:{}'.format(lport))
+            if "CMIS_REINIT_REQUIRED" not in app_port_table_fvs_dict:
+                appl_db.set(appl_db.APPL_DB, 'PORT_TABLE:{}'.format(lport), "CMIS_REINIT_REQUIRED", "true")
+                self.log_notice("Added CMIS_REINIT_REQUIRED for lport {}".format(lport))
+            if "NPU_SI_SETTINGS_SYNC_STATUS" not in app_port_table_fvs_dict:
+                appl_db.set(appl_db.APPL_DB, 'PORT_TABLE:{}'.format(lport), "NPU_SI_SETTINGS_SYNC_STATUS", "NPU_SI_SETTINGS_DEFAULT")
+                self.log_notice("Added NPU_SI_SETTINGS_SYNC_STATUS for lport {}".format(lport))
+
+        self.log_notice("XCVRD INIT: Port init fields initialized in APP_PORT_TABLE")
+
+    """
+    Subscribes APPL_DB PORT_TABLE table events
+    """
+    def subscribe_appl_port_table(self):
```

Comment:
there is an existing function  `wait_for_port_config_done(self, namespace)` that has the same logic, suggest reducing the duplicated code by calling this new function in that one.

Reply:
I have now removed the duplicate code and xcvrd is now subscribing only once for APPL_DB updates